clang/lib/Sema/TreeTransform.h

```diff
@@ -4661,6 +4718,42 @@ TreeTransform<Derived>::TransformTemplateName(CXXScopeSpec &SS,
                                             ObjectType, AllowInjectedClassName);
   }
 
+  if (DeducedTemplateStorage *DTN = Name.getAsDeducedTemplateName()) {
+    TemplateName Underlying = DTN->getUnderlying();
+    TemplateName TransUnderlying = getDerived().TransformTemplateName(
+        SS, Underlying, NameLoc, ObjectType, FirstQualifierInScope,
+        AllowInjectedClassName);
+    if (TransUnderlying.isNull())
+      return TemplateName();
+
+    DefaultArguments DefArgs = DTN->getDefaultArguments();
+
+    TemplateArgumentListInfo TransArgsInfo;
+    using Iterator =
+        TemplateArgumentLocInventIterator<Derived, TemplateArgument *>;
```

Comment:
```suggestion
    using Iterator =
        TemplateArgumentLocInventIterator<Derived, const TemplateArgument *>;
```

I think you might be able to get rid of the `const_cast`